src/ui/buttons/buttons.jsx

```diff
@@ -150,6 +152,47 @@ export function validateButtonProps(props: ButtonPropsInputs) {
   normalizeButtonProps(props);
 }
 
+function calculateMsgPosition({
+  message,
+  tagline,
+  layout,
+  fundingSources,
+  fundingSource,
+}): string {
+  if (!message) {
+    return null;
+  }
+  const { position } = message;
+  // Cases for throwing an error
+  if (tagline && layout === BUTTON_LAYOUT.HORIZONTAL && !fundingSource) {
+    throw new ValidationError("Button cannot show both Message and Tagline.");
+  }
+  if (
+    fundingSources.indexOf(FUNDING.CARD) !== -1 &&
+    position === MESSAGE_POSITION.BOTTOM
```

Comment:
Similar comment as above but for something like `showPoweredBy`. Also consider updating `fundingSources.indexOf(FUNDING.CARD) !== -1` to `fundingSources.includes(FUNDING.CARD)` to align with best practices.

Reply:
on second look, i specified `fundingSources.includes(FUNDING.CARD)` here intentionally to cover both PoweredBy and this condition from the story's notes:

- Revert to top position if merchant puts bottom when the Debit or Credit card button is shown (this button already displays a message below it)

Reply:
Should we be accounting for the `layout === BUTTON_LAYOUT.VERTICAL` of the `showPoweredBy`? Would we want `showPoweredBy && position === MESSAGE_POSITION.BOTTOM`?

Reply:
you were right after all - fixed